File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/hashx/registry.go
package hashx

// Registry lists every checksum algorithm in catalogue order.
func Registry() []Algorithm {
	return []Algorithm{
		{Name: "fnv1a32", Apply: Fnv1a32},
		{Name: "fnv1a64", Apply: Fnv1a64},
		{Name: "djb2", Apply: Djb2},
		{Name: "sdbm", Apply: Sdbm},
		{Name: "elf_hash", Apply: ElfHash},
		{Name: "adler32", Apply: Adler32},
		{Name: "fletcher16", Apply: Fletcher16},
		{Name: "fletcher32", Apply: Fletcher32},
		{Name: "crc32_ieee", Apply: Crc32Ieee},
		{Name: "crc32c", Apply: Crc32c},
		{Name: "crc16_ccitt", Apply: Crc16Ccitt},
		{Name: "crc8_atm", Apply: Crc8Atm},
		{Name: "jenkins_oaat", Apply: JenkinsOaat},
		{Name: "murmur3_32", Apply: Murmur332},
		{Name: "xor_rotate", Apply: XorRotate},
		{Name: "bsd_sum16", Apply: BsdSum16},
	}
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/hash/registry.cpp
#include "freight/hashes.h"

namespace freight {

const std::vector<HashAlgorithm>& hashRegistry() {
  static const std::vector<HashAlgorithm> registry = {
      HashAlgorithm{"fnv1a32", &hash_fnv1a32},
      HashAlgorithm{"fnv1a64", &hash_fnv1a64},
      HashAlgorithm{"djb2", &hash_djb2},
      HashAlgorithm{"sdbm", &hash_sdbm},
      HashAlgorithm{"elf_hash", &hash_elf_hash},
      HashAlgorithm{"adler32", &hash_adler32},
      HashAlgorithm{"fletcher16", &hash_fletcher16},
      HashAlgorithm{"fletcher32", &hash_fletcher32},
      HashAlgorithm{"crc32_ieee", &hash_crc32_ieee},
      HashAlgorithm{"crc32c", &hash_crc32c},
      HashAlgorithm{"crc16_ccitt", &hash_crc16_ccitt},
      HashAlgorithm{"crc8_atm", &hash_crc8_atm},
      HashAlgorithm{"jenkins_oaat", &hash_jenkins_oaat},
      HashAlgorithm{"murmur3_32", &hash_murmur3_32},
      HashAlgorithm{"xor_rotate", &hash_xor_rotate},
      HashAlgorithm{"bsd_sum16", &hash_bsd_sum16},
  };
  return registry;
}

}  // namespace freight
